src/components/pages/profiles/UserProfile.js

```diff
@@ -0,0 +1,203 @@
+/* eslint-disable no-nested-ternary */
+import React, { Component } from 'react';
+import FormData from 'form-data';
+import { connect } from 'react-redux';
+import {
+  Container, Row, Button, Col, Form,
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import Breadcrumbs from '../../global/Breadcrumbs';
+import AlertComponent from '../../global/AlertComponent';
+import { GetUserProfile, UpdateUserProfile } from '../../../actions/profileAction';
+
+export class UserProfile extends Component {
+  state = {
+    username: '',
+    gender: '',
+    phone: '',
+    language: '',
+    country: '',
+    company: '',
+    department: '',
+    isLoading: false,
+    selectedFile: '',
+  };
+
+  async componentDidMount() {
+    const { props } = this;
+    this.setState({ isLoading: true });
+    await props.GetUserProfile();
+    const { data } = this.props;
+    this.setState({
+      username: data.profile.username,
+      gender: data.profile.gender,
+      phone: data.profile.phone,
+      language: data.profile.language,
+      country: data.profile.country,
+      company: data.profile.company,
+      department: data.profile.department,
+      image: data.profile.image,
+      isLoading: false,
+    });
+  }
+
+  handleChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value,
+    });
+  };
+
+
+    handleImage = (event) => {
+      this.setState({ selectedFile: event.target.files[0] });
+    }
+
+  handleSubmit = async (e) => {
+    const { props } = this;
+    e.preventDefault();
+
+    const {
+      username, gender, phone, language, country, company, department, selectedFile,
+    } = this.state;
+    this.setState((state) => ({ ...state, isLoading: true }));
+    const form = new FormData();
+    form.append('username', username);
+
+    if (gender) {
+      form.append('gender', gender);
+    }
+    if (phone) {
+      form.append('phone', phone);
+    }
+    if (language) {
+      form.append('language', language);
+    }
+    if (country) {
+      form.append('country', country);
+    }
+    if (company) {
+      form.append('company', company);
+    }
+    if (department) {
+      form.append('department', department);
+    }
+    if (selectedFile) {
+      form.append('image', selectedFile);
```

Comment:
These are if statements are too many there is a better way of validating the fields reach out to @alainmateso he worked on something similar